File: src/main/java/net/arvandor/numinoustreasury/listener/PlayerItemConsumeListener.java
package net.arvandor.numinoustreasury.listener;

import net.arvandor.numinoustreasury.item.NuminousItemStack;
import net.arvandor.numinoustreasury.item.action.NuminousOnEat;
import org.bukkit.event.EventHandler;
import org.bukkit.event.Listener;
import org.bukkit.event.player.PlayerItemConsumeEvent;

import java.util.List;

public final class PlayerItemConsumeListener implements Listener {

    @EventHandler
    public void onPlayerItemConsume(PlayerItemConsumeEvent event) {
        NuminousItemStack numinousItemStack = NuminousItemStack.fromItemStack(event.getItem());
        if (numinousItemStack == null) return;
        List<NuminousOnEat> onEat = numinousItemStack.getItemType().getOnEat();
        for (NuminousOnEat action : onEat) {
            action.onEat(event);
        }
    }

}

File: src/main/kotlin/net/arvandor/numinoustreasury/listener/PlayerItemConsumeListener.kt
package net.arvandor.numinoustreasury.listener

import net.arvandor.numinoustreasury.item.NuminousItemStack
import org.bukkit.event.EventHandler
import org.bukkit.event.Listener
import org.bukkit.event.player.PlayerItemConsumeEvent

class PlayerItemConsumeListener : Listener {
    @EventHandler
    fun onPlayerItemConsume(event: PlayerItemConsumeEvent) {
        val numinousItemStack: NuminousItemStack = NuminousItemStack.Companion.fromItemStack(event.item) ?: return
        val onEat = numinousItemStack.itemType.onEat
        for (action in onEat) {
            action.onEat(event)
        }
    }
}
